test/e2e/common/node/pod_admission.go

```diff
@@ -65,6 +66,49 @@ var _ = SIGDescribe("PodOSRejection", framework.WithNodeConformance(), func() {
 	})
 })
 
+var _ = SIGDescribe("PodOSRejectionStatus", func() {
+	f := framework.NewDefaultFramework("pod-os-rejection-status")
+	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
+	ginkgo.Context("Kubelet", func() {
+		ginkgo.It("should reject pod when the node OS doesn't match pod's OS", func(ctx context.Context) {
+			linuxNode, err := findLinuxNode(ctx, f)
+			framework.ExpectNoError(err)
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wrong-pod-os",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: v1.PodSpec{
+					OS: &v1.PodOS{
+						Name: "windows", // explicitly set the pod OS to a wrong but valid value
+					},
+					Containers: []v1.Container{
+						{
+							Name:  "wrong-pod-os",
+							Image: imageutils.GetPauseImageName(),
+						},
+					},
+					NodeName: linuxNode.Name, // Set the node to an node which doesn't support
```

Comment:
you need the Pod status before it was scheduled and after it was rejected by the kubelet. So maybe create without the node name first, save the status, set the node name, then compare statuses

Reply:
> you need the Pod status before it was scheduled and after it was rejected by the kubelet. So maybe create without the node name first, save the status,

```go
pod = e2epod.NewPodClient(f).Create(ctx, pod)
```
the above function will return a status which the pod is not observed by kubelet even if I set the node name for the pod.

Reply:
ok. I wasn't sure how `Create` works. Can we ever get any race conditions there? I have trouble trusting the behavior of Create here and it will be more reliable easier to understand if we will make test more explicit like I suggested

Reply:
@SergeyKanzhelev I get the pod via the watch channel and compare the status between old and new object. It can work well even if other programs in the cluster change the pod status.